lib/QueueArray.h

```diff
@@ -0,0 +1,316 @@
+/*
+ *  QueueArray.h
```

Comment:
So I initially was using that to try to build my DataPacket so the message could be of variable length, but that added too much complexity that got eliminated one I just moved to a fixed-size packet. I'll remote it with this next commit :)